defi/lending_platform/src/lending_platform.rs

```diff
@@ -0,0 +1,436 @@
+use sbor::*;
+use scrypto::prelude::*;
+
+blueprint! {
+    struct LendingPlatform {
+        assets: LazyMap<Address, Vault>,
+        asset_ltv_ratios: LazyMap<Address, Decimal>,
+        loan_balances: LazyMap<Address, Decimal>,
+        users: LazyMap<Address, User>,
+        admin_badge: ResourceDef
+    }
+
+    impl LendingPlatform {
+        fn calculate_total_collateral(&self, user: &User) -> Decimal {
```

Comment:
Can't you keep track of the `total_collateral` and `total_loan_balance` inside variables on the User struct ? That way you don't have to loop each time

Reply:
I don't believe so. The reason is that in the future we will be using a price oracle that will provide the current prices of each asset which will need to be taken into account each time we need access the `total_collateral` or `total_loan_balance`.

Additionally, if we were to compute theses values on the User level, would would have to pass asset prices or an oracle into the User object which would in opinion break the domain boundaries of a `User`